src/main/java/strategy/PurchaseOperation.java

```
@@ -0,0 +1,18 @@
+package strategy;
+
+import core.basesyntax.FruitTransaction;
+import db.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
```

The task requires handling invalid input, including negative quantities. You should add a check here to ensure `transaction.getQuantity()` is not negative. A purchase with a negative quantity is an invalid operation and should throw an exception.